src/main/java/org/terasology/projsndwv/systems/ApiarySystem.java

```diff
@@ -15,23 +15,273 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(ApiarySystem.class);
+
+    private Genome genome;
+
+    /**
+     * The delayed action id for life ticks.
+     */
+    public static final String LIFE_TICK_EVENT = "life_tick";
+
+    /**
+     * The delayed action id for the completion of mating.
+     */
+    public static final String MATING_EVENT = "mating";
+
+
+
+    /**
+     * The slot index for the princess/queen bee slot in an apiary's inventory.
+     */
+    public static final int SLOT_FEMALE = 0;
+
+    /**
+     * The slot index for the drone bee slot in an apiary's inventory.
+     */
+    public static final int SLOT_MALE = 1;
+
+    /**
+     * A list of slot indices for the otuput slots in an apiary's inventory.
+     */
+    public static final List<Integer> SLOTS_OUT = Collections.unmodifiableList(Arrays.asList(2, 3, 4, 5, 6, 7, 8));
+
+
+
+    /**
+     * The locus in a bee's genetics indicating the species of a bee.
+     */
+    public static final int LOCUS_SPECIES = 0;
+
+    /**
+     * The locus in a bee's genetics indicating the speed at which a bee's life ticks down.
+     */
+    public static final int LOCUS_SPEED = 1;
+
+    /**
+     * The locus in a bee's genetics indicating the length of a bee's lifespan.
+     */
+    public static final int LOCUS_LIFESPAN = 2;
+
+    /**
+     * The locus in a bee's genetics indicating the number of drone offspring a bee will have.
+     */
+    public static final int LOCUS_OFFSPRING_COUNT = 3;
+
+    /**
+     * The time, in milliseconds, that mating takes in an apiary.
+     */
+    public static final long MATING_TIME = 1000L;
+
+    @ReceiveEvent
+    public void beforeItemPutIntoApiary(BeforeItemPutInInventory event, EntityRef entity, ApiaryComponent component) {
+        if (event.getSlot() == SLOT_FEMALE) {
+            if (!event.getItem().hasComponent(BeeComponent.class) || event.getItem().getComponent(BeeComponent.class).type == BeeComponent.BeeType.DRONE) {
+                event.consume();
+            }
+        }
+        else if (event.getSlot() == SLOT_MALE) {
+            if (!event.getItem().hasComponent(BeeComponent.class) || event.getItem().getComponent(BeeComponent.class).type != BeeComponent.BeeType.DRONE) {
+                event.consume();
+            }
+        }
+        else if (event.getInstigator() != entity) {
+            event.consume();
+        }
+    }
+
     @ReceiveEvent
-    public void onApiaryInitialized(OnAddedComponent event, EntityRef entity, ApiaryComponent component) {
-        if (!entity.hasComponent(InventoryComponent.class)) {
-            InventoryComponent inventoryComponent = new InventoryComponent(9);
-            inventoryComponent.privateToOwner = false;
-            entity.addComponent(inventoryComponent);
+    public void onApiaryInventoryChanged(InventorySlotChangedEvent event, EntityRef entity, ApiaryComponent component) {
+        DelayManager delayManager = CoreRegistry.get(DelayManager.class);
```

Comment:
You have `@In`ed the delay manager above, why are you fetching it from core registry here?